sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/JsonSerializable.java

```diff
@@ -672,4 +689,22 @@ public String toString() {
     public ObjectNode getPropertyBag() {
         return this.propertyBag;
     }
+
+    <T> boolean containsJsonSerializable(Class<T> c) {
+        return CompositePath.class.isAssignableFrom(c)
```

Comment:
You should be able to do class equality here as none of the classes is expected to be subclassed.

```suggestion
CompositePath.class.equals(c)
```

Reply:
changed to use equals except Index class.